lib/rubocop/cop/util.rb

```diff
@@ -187,7 +187,7 @@ def directions(side)
 
       def begins_its_line?(range)
         source_before_end = range.source_buffer.source[0...range.begin_pos]
-        source_before_end =~ /\n\s*\Z/
+        source_before_end.reverse!.sub!(/\A[ \t\r\f]*/, '').start_with?("\n")
```

Comment:
This should definitely have some explanations, otherwise someone might "optimize" it in the future. :-) Maybe taking a look at the implementation of `lstrip` might be useful as well, as I guess it's efficient, but kills newlines.